src/main/java/ca/spottedleaf/moonrise/mixin/blockstate_propertyaccess/StateHolderMixin.java

```diff
@@ -141,4 +152,24 @@ public <T extends Comparable<T>> T getValue(final Property<T> property) {
     public <T extends Comparable<T>> boolean hasProperty(final Property<T> property) {
         return property != null && this.optimisedTable.hasProperty(property);
     }
+
+    /**
+     * @reason Replace with optimisedTable
+     * @author embeddedt
+     */
+    @Overwrite
+    public Collection<Property<?>> getProperties() {
+        return this.optimisedTable.getProperties();
+    }
+
+    /**
+     * @reason Replace with optimisedTable
+     * @author embeddedt
+     */
+    @Overwrite
+    public Map<Property<?>, Comparable<?>> getValues() {
+        ZeroCollidingReferenceStateTable<O, S> table = this.optimisedTable;
+        // We have to use this.values until the table is loaded
+        return table.isLoaded() ? table.getMapView(this.tableIndex) : this.values;
```

Comment:
table may be used as a map for property -> comparable even when not loaded, as the property index is initialised

Reply:
In my testing, the table relies on `getValues` to initialize itself. If `getValues` unconditionally delegates to the table it ends up just not having any properties.